include/aos/common/tools/time.hpp

```diff
@@ -17,15 +17,161 @@
 
 namespace aos {
 
+/**
+ * Size of a time in string representation.
+ */
+const auto cTimeStrLen = AOS_CONFIG_TIME_STR_LEN;
+
 /**
  * Base type for a time duration in nanoseconds. Can also be negative to set a point back in time.
  */
-using Duration = int64_t;
+class Duration {
+public:
+    /**
+     * Default constructor.
+     */
+    constexpr Duration() = default;
+
+    // cppcheck-suppress noExplicitConstructor
+    /**
+     * Constructs object instance.
+     *
+     * @param duration duration in nanoseconds.
+     */
+    constexpr Duration(int64_t duration)
+        : mDuration(duration)
+    {
+    }
+
+    /**
+     * Returns duration in nanoseconds.
+     *
+     * @result int64_t.
+     */
+    constexpr int64_t Count() const { return mDuration; }
```

Comment:
It is not obvious that Count is in nanoseconds. So I propose to have such method as:

```c++
uint64_t Nanoseconds() - return duration in nanoseconds
uint64_t Microseconds() - return duration in microseconds
uint64_t Milliseconds() - return duration in milliseconds
float Seconds() - return duration in float seconds
float Minutes() - return duration in float minutes
float Hours() - return duration in float hours
```